README.md

```diff
@@ -1,8 +1,118 @@
+
+
+
+* Chapter 4:
+* Chapter 5:
+* Chapter 6:
+* Chapter 7:
+* Chapter 8:
+* Chapter 9:
+* Chapter 10:
+* Chapter 11:
+* Chapter 12:
+* Chapter 13:
+* Chapter 14:
+* Chapter 15:
+
+TomcatEE 9 = TomEE 9 plus
+WildFly 6 = Version 26.1.2
+
+# Building
+
+To Build targeting Derby Db
+
+    mvn clean package
+
+There are other build options described [later](#other-build-options)
+
+# IDE Usage
+Import the pom.xml into your IDE.
+
+
+# Goal: It Just works
+To provide a 'it just works' experience for myself that others should be able to use.
+This means:
+* Required libraries will be downloaded
+* DB will be run in memory to remove setup and installation effort whenever possible
+* DDL, Tables, will be auto created
+* Generated files will be ignored by Git
+
+## Other changes
+* Target Java 17 - But should build under Java 11 if desired.
+* Ability to switch between derby and hsqldb
+
+## Other build options
+By default, *derby* is used as the underlying Database.
+
+    mvn clean package
+
+If you want to use *hsqldb* , activate the *hsqldb* maven profile. 
+
+    mvn -P hsqldb clean package
+
+During this process it was useful to use hsqldb initially to get it to work due to its better documentation and adoption.
+then switch back to derby afterwards. Similarly, H2 is also a good alternate db for learning
+due to its Spring Boot connection.
+
+## Maven vs Gradle
+While I have used both, maven by its very design is a constrained, somewhat inflexible build system out of the box.
+However, that inflexibility by its very nature makes debugging problems straight forward, predictable, 
+and hence its use. In addition, while IDE's can run/build Gradle projects using gradle, the use of maven allows your
+IDE to convert a project into its internal build system, which is an advantage.
+
+## Original book environment
+
+**Code Organization**
+
+The book used ANT patterns such as separating logically different groups of code, architectural layers,  
+such as the business model, client, servlet, into different directory trees yet still under the same build directory.
+In real applications these would be libraries or services. That layout while convenient,  is tractable as organized.
+We could use a maven plugins such as *build-helper-maven-plugin* to retain the existing directory structure. However,  
+IDE's have a limited support of 3rd party plugins when importing. The easiest alternatives are:
+1) Combine source directories together for convenience under src/main/java.
+2) Use separate maven modules for each layer
+
+The former approach most closely matches the authors original intent to show a layered architecture, and will be used.
+
+**Unknown dependency versions**
+
+With no dependency management system and JARS checked into Git, knowing what versions are used is very time consuming.
```

Comment:
whatever version currently provided by the server is used and one can find it either in the server log, in jar's manifest or, in some cases, by running `java -jar ...`. Manually updating server-side version of the library to really match the one set up in the dependency management is - in some cases - a non-trivial task going way beyond the scope of these samples.

Reply:
Understood, and most applications will be building against the API's not specific implementations whenever possible.

FYI: At some point I will create a branch w/o maven to contribute a few fixes for example: Chapter 3 Example 08  uses EJB Context  lookup , but states it uses InitialContext lookup. I made changes so that it used  InitialContext lookup as the book intended, since Example 09 already demonstrates EJB Context  lookup.